src/instructlab/sdg/pipeline.py

```diff
@@ -67,17 +94,21 @@ def generate(self, dataset) -> Dataset:
         dataset: the input dataset
         """
         for block_prop in self.chained_blocks:
+            # Parse and instantiate the block
             block_name = block_prop["name"]
             block_type = _lookup_block_type(block_prop["type"])
             block_config = block_prop["config"]
             drop_columns = block_prop.get("drop_columns", [])
             drop_duplicates_cols = block_prop.get("drop_duplicates", False)
             block = block_type(self.ctx, self, block_name, **block_config)
```

Comment:
I know this contradicts what @russellb said, but I think it would be good to wrap the instantiation too - e.g. the config_path not found error I showed in the issue

(I agree with not wrapping the whole contents of this for loop though)